src/interface/src/app/services/plan.service.ts

```diff
@@ -314,10 +315,10 @@ export class PlanService {
   }
 
   /** Fetches a scenario by its id from the backend. */
-  getScenario(scenarioId: string): Observable<Scenario> {
+  getScenario(scenarioId: number): Observable<Scenario> {
     const url = BackendConstants.END_POINT.concat(
       '/planning/get_scenario_by_id/?id=',
```

Comment:
you can use string interpolation here to clean up this a bit
`/planning/get_scenario_by_id/?id=${scenarioId}`

Reply:
I may save that for a future pr since I'll likely go through and update all the service calls to be formatted like that

Reply:
Added  a todo!